components/Anchor.tsx

```diff
@@ -0,0 +1,39 @@
+import Link from 'next/link'
+import {
+  AnchorHTMLAttributes,
+  ComponentType,
+  DetailedHTMLProps,
+  Fragment,
+} from 'react'
+
+export type AnchorProps<T = HTMLAnchorElement> = DetailedHTMLProps<
+  AnchorHTMLAttributes<T>,
+  T
+>
+
+/**
+ * Adaptive link component that can be used for both relative Next.js pages and
+ * external links, recommended for sidebar and footer links
+ *
+ * @see https://nextjs.org/docs/api-reference/next/link
+ */
+const Anchor = (props: AnchorProps) => {
+  const isRelative = props.href?.startsWith('/') ?? false
+  const Wrap = (isRelative ? Link : Fragment) as ComponentType
+
+  // if it's a relative link, we need to pass href so it can navigate client-side
+  const wrapProps = isRelative ? { href: props.href } : {}
+
+  // if it's an external link, passing the target="_blank" prop will open externally
+  const linkProps = !isRelative
+    ? { target: '_blank', rel: 'noopener noreferrer' }
+    : {}
```

Comment:
Curious how this is different from `Link`'s `isExternal` boolean 🤔

Reply:
`Link` is a Next.js component (not Chakra UI), so there's no `isExternal` prop. But Chakra UI's `Link#isExternal` prop behaves the same like this.

Reply:
Ah! Good shout 😅